### DISCUSSION TITLE
Is it possible to administrate client attributes in the Keycloak admin UI?

### DISCUSSION THREAD

**Discussion by Amiron49** [keycloak/keycloak#9626] [Q&A] [answered]
Is there any way to set or see openid-connect client attributes in the admin UI? The rest API "ClientRepresentation" does have an attribute map, so I believe that clients do have attributes... right? Or am I blind?

Just in case I'm using the wrong solution for my problem: 

In the deployment I configured an attribute mapper to map a special attribute to the claims of the jwt. That works well for user, but I also need that value in my client service-accounts jwts. 

Alternative solutions I considered: 
- Create a role per client with the attribute and assign it to the service-account -> BUT mapping role attributes seems to be only possible with js scripts, which looks a bit wonky.
- Create a scope per client that attaches the id -> Also not ideal but feels better than the role solution

**Comment by Amiron49** [comment, marked as answer]
Of course, once I make a discussion I find a related pull request #6250. (It's not currently possible)

Super late Edit:
Just an update for anybody that may enounter this in the future. 

My actual problem was "give service accounts attributes" which is not solved by giving the client attributes. The service accounts cannot be directly accessed in the User search but if you know the underlying Id you can manually browse to the edit page with the admin ui. 

Ultimately I added the extra infromation I needed there.

**Reply by bernhof** [reply]
Unfortunately this workaround excludes public clients since they cannot have service accounts.

**Comment by jbman** [comment]
We would also welcome a tab which displays client attributes in the admin console and allows to add or update an attribute. 
Keycloak also uses client attributes for some client settings e.g. `oauth2.device.polling.interval`. Such entries which have an existing switch in the console should be shown as read only.

Same approach should be applied to realm attributes.

**Comment by ataylormays-owner** [comment]
Any updates on this? Consider this some community appetite for the feature here in 2026, in addition to the upvotes on the original question and comments above!

**Comment by bernhof** [comment]
Relevant: https://github.com/keycloak/keycloak/discussions/24805#discussioncomment-16478441
